plugin/trino-teradata/src/test/java/io/trino/plugin/integration/AbstractTeradataJDBCTest.java

```diff
@@ -0,0 +1,110 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.trino.plugin.integration;
+
+import io.trino.testing.AbstractTestQueryFramework;
+import io.trino.testing.QueryRunner;
+import io.trino.testing.datatype.CreateAndInsertDataSetup;
+import io.trino.testing.datatype.DataSetup;
+import org.junit.jupiter.api.AfterAll;
+import org.junit.jupiter.api.BeforeAll;
+
+/**
+ * Abstract base class for Teradata JDBC integration tests in Trino.
+ * <p>
+ * This class initializes a Teradata test database using a provided {@code databaseName},
+ * manages its lifecycle, and provides support for creating and inserting data
+ * via {@link DataSetup}.
+ * </p>
+ *
+ * <p>Derived classes must implement {@link #initTables()} to set up the schema
+ * and tables needed for testing.</p>
+ *
+ * <p>Database is created before all tests and dropped after all tests. A unique
+ * {@code databaseName} can be passed by subclasses to allow isolated parallel test runs.</p>
+ */
+abstract class AbstractTeradataJDBCTest
```

Comment:
Merge into TestTeradataJDBCTypeMapping.

Reply:
addressed in new PR https://github.com/trinodb/trino/pull/26731